lib/mongo.ex

```diff
@@ -167,7 +167,7 @@ defmodule Mongo do
     * `:max_time` - Specifies a time limit in milliseconds
     * `:use_cursor` - Use a cursor for a batched response (Default: true)
   """
-  @spec aggregate(GenServer.server(), collection, [BSON.document()], Keyword.t()) :: cursor
+  @spec aggregate(GenServer.server(), collection, [BSON.document()], Keyword.t()) :: cursor | {:error, Mongo.Error.t()}
```

Comment:
I would say it should return `{:ok, cursor()} | {:error, Mongo.Error.t()}` to make API more consistent.

Reply:
I agree but I think this also depends on how far we want to go with the interface changes.  Eventually, these changes may impact the existing user of the API.

Reply:
We can release new major version. If we change API then we should make it right to not change it again and again.

Reply:
So I shall return  {:ok, cursor()}   ?

Reply:
Yes, I think so.